docs/src/pages/components/pagination/PaginationLink2.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { MemoryRouter as Router } from 'react-router';
+import { Link as RouterLink } from 'react-router-dom';
+import { makeStyles } from '@material-ui/core/styles';
+import { usePagination } from '@material-ui/lab/Pagination';
+import Pagination from '@material-ui/lab/Pagination';
+import PaginationItem from '@material-ui/lab/PaginationItem';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    '& > *': {
+      marginTop: theme.spacing(2),
+    },
+  },
+}));
+
+// The use of React.forwardRef will no longer be required for react-router-dom v6.
+// See https://github.com/ReactTraining/react-router/issues/6056
+const Link = React.forwardRef((props, ref) => <RouterLink innerRef={ref} {...props} />);
+
+export default function PaginationLink2() {
+  const { items } = usePagination({
+    count: 10,
+  });
+
+  return (
+    <Router>
+      <Pagination>
+        {items.map((item, index) => (
+          <li>
```

Comment:
No, it was moved up from PaginationItem to accomodate this API.

Reply:
OK, it took me a bit of time to understand. I think that I get it now. thanks. It's because it uses the hook for the logic and the component for the UI rendering. I think that using custom styled components pagination elements, as a potentially customization approach could remove the confusion.